addons/mrp/models/mrp_workorder.py

```diff
@@ -737,6 +737,17 @@ def _compute_qty_remaining(self):
         for wo in self:
             wo.qty_remaining = max(float_round(wo.qty_production - wo.qty_reported_from_previous_wo - wo.qty_produced, precision_rounding=wo.production_id.product_uom_id.rounding), 0)
 
+    def _get_product_expected_duration(self, alternative_workcenter=False):
+        """In case there are specific capacities defined in the workcenter
+        that matches the product we are producing. Add the extra-time.
+        """
+        self.ensure_one()
+        workcenter = alternative_workcenter or self.workcenter_id
+        for product in workcenter.capacity_ids:
+            if product.product_id == self.product_id:
+                return product.time_start + product.time_stop
+        return 0.0
+
```

Comment:
It's better to have this in the `mrp.workcenter` since we are looking at the product expected duration that is specific to a workcenter, not to a workorder. It would also be better to make this method more generic when you move it too (you won't need `alternative_workcenter=False` since you can call it directly on the workcenter object). Also look at `_get_capacity` to see how you can simplify your code down to 2 lines.

Reply:
Changes added in 09a79a9:
- move the def to work_center
- shorten the def